src/launchpad/sentry_client.py

```diff
@@ -186,6 +255,122 @@ def download_artifact_to_file(self, org: str, project: str, artifact_id: str, ou
+                                chunk_downloaded += len(data_chunk)
+
+                                # Log progress every second during chunk download
+                                current_time = time.time()
+                                if current_time - last_log_time >= 1.0:
+                                    elapsed = current_time - chunk_start_time
+                                    self._log_progress(
+                                        chunk_downloaded, expected_chunk_size, elapsed, f"  Chunk {chunk_num} "
+                                    )
+                                    last_log_time = current_time
+
+                        chunk_data = b"".join(chunk_data_parts)
+                        request_time = time.time() - chunk_start_time
+
+                        if len(chunk_data) != expected_chunk_size:
+                            raise SentryClientError(
+                                detail=f"Incomplete chunk: got {len(chunk_data)} bytes, expected {expected_chunk_size}"
+                            )
+
+                        out.write(chunk_data)
+                        downloaded += len(chunk_data)
+
+                        # Log completion of this chunk with timing
+                        speed_mbps = chunk_mb / request_time if request_time > 0 else 0
+                        progress = (downloaded / total_size) * 100
+                        logger.info(
+                            f"Completed chunk {chunk_num}/{total_chunks} in {request_time:.1f}s ({speed_mbps:.1f} MB/s) - Total: {downloaded / MB_DIVISOR:.2f} MB ({progress:.1f}%)"
+                        )
+
+                        chunk_success = True
+                        break  # Success, move to next chunk
+                    else:
+                        raise SentryClientError(
+                            detail=f"Unexpected status code: {response.status_code}", response=response
+                        )
+
+                except Exception as e:
+                    if chunk_attempt < 2:
+                        wait_time = 2**chunk_attempt  # Exponential backoff
+                        logger.warning(
+                            f"Chunk download failed (attempt {chunk_attempt + 1}/3), retrying in {wait_time}s: {e}"
+                        )
+                        time.sleep(wait_time)
+                        continue
+                    else:
+                        logger.error(f"Failed to download chunk after 3 attempts: {e}")
+                        # Don't raise immediately, let's try the fallback
+                        break
+
+            if not chunk_success:
+                logger.error(f"Failed to download chunk at byte {start}, falling back to standard download")
+                # Reset file position and try standard download from beginning
+                out.seek(0)
+                out.truncate()
+                return self.download_artifact_to_file(org, project, artifact_id, out)
```

Comment:
I don't think it is worth supporting both. It gets quite complicated it seems fine to mandate the endpoint supports Range (once you land the Sentry side PR). We could even support both in simpler way:
- Try and download everything
- If that fails and endpoint supports Range -> use Range to resume the download
- If that fails and the endpoint does not not support Range -> raise

That way we still support non-range - we just lose the retrying and if you want the retrying you should support Range.
   - use range qury